src/dtx/dtx_common.c

```diff
@@ -492,6 +492,12 @@ dtx_aggregation_pool(struct dss_module_info *dmi, struct dtx_batched_pool_args *
 		if (dbca->dbca_agg_gen == tls->dt_agg_gen)
 			break;
 
+		if (unlikely(dbca->dbca_deregister)) {
+			/* Someone is closing current container. */
+			D_ASSERT(d_list_empty(&dbca->dbca_pool_link));
+			continue;
+		}
```

Comment:
Looks it could race with dtx_free_dbca()? (both could call sched_req_put(dbca->dbca_agg_req)).

     if (dbca->dbca_agg_req != NULL) {
                if (!dbca->dbca_agg_done)
                        sched_req_wait(dbca->dbca_agg_req, true);
                /* Just to be safe... */
                if (dbca->dbca_agg_req != NULL) {
                        D_ASSERT(dbca->dbca_agg_done);
                        sched_req_put(dbca->dbca_agg_req);
                        dbca->dbca_agg_req = NULL;
                        dbca->dbca_agg_done = 0;
                }
        }

Reply:
Yes, I am further working on the patch with reference.